webserver/login/__init__.py

```diff
@@ -33,6 +34,21 @@ def load_user(user_id):
     else:
         return None
 
+@login_manager.request_loader
+def load_user(request):
+    key = request.headers.get("Authorization")
+    user = None
+    if key:
+        parts = key.split(" ")
+        if len(parts) == 2 and parts[0] == "Token":
+            user = db.user.get_by_api_key(parts[1])
+        else:
+            raise Unauthorized
+    if user:
+        return User.from_dbrow(user)
+    else:
+        return None
```

Comment:
It does, as I understand flask will either use its normal session auth, then if that doesn't work it'll use this (for the whole site). It shouldn't affect users browsing the main site (e.g. dataset page), since they won't be sending a header.
We could take a look at see if you can restrict this loader to just a particular blueprint, but I like this method better than a standard `@auth_required` decorator because it seamlessly integrates the `current_user` in the same way as normal browser auth.